fvtest/compilertriltest/MaxMinTest.cpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "OpCodeTest.hpp"
+#include "default_compiler.hpp"
+
+int32_t imax(int32_t l, int32_t r) {
+  return l > r ? l : r;
+}
+
+int32_t imin(int32_t l, int32_t r) {
+  return l < r ? l : r;
+}
+
+int64_t lmax(int64_t l, int64_t r) {
+  return l > r ? l : r;
+}
+
+int64_t lmin(int64_t l, int64_t r) {
+  return l < r ? l : r;
+}
+
+std::vector<int64_t> longMaxMinTestValues()
+   {
+   std::vector<int64_t> longValues =TRTest::const_values<int64_t>();
+   longValues.push_back(0x80000000L);
+   return longValues;
+   }
+
+class Int32MaxMin : public TRTest::BinaryOpTest<int32_t> {};
+
+TEST_P(Int32MaxMin, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int32 (block (ireturn (%s (iconst %d) (iconst %d)) )))", param.opcode.c_str(), param.lhs, param.rhs);
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int32_t (*)(void)>();
+    volatile auto exp = param.oracle(param.lhs, param.rhs);
+    volatile auto act = entry_point();
```

Comment:
Do these really need to be volatile? I think just having `ASSERT_EQ(eparam.oracle(param.lhs, param.rhs), entry_point())` should work.